### PR TITLE
feat: add hashed of original id(s)when id(s) are present in body

### PR DESCRIPTION
**Pull request by ilhaoni**
This is to support the case whereby we would still like to retrieve the pre-transformed id and forward it as well. 

Currently, we only pass the transformed ids forward

Test and review together with [PR34](https://github.com/rationally-app/backend-api/pull/34)

### CODE REVIEW

File: src/functionHandlers/proxy/transformers/bodyTransformer.ts [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
   const { id, ids } = body;
   if (id) {
     body.id = await transformer(id);
+    body.hashedOriginalId = identifierStringToSaltedHash(id);
   }
   if (ids) {
     const deferredTransformedIds = ids.map(transformer);
     body.ids = await Promise.all(deferredTransformedIds);
+
+    const hashedIds = ids.map(identifierStringToSaltedHash);
```
Comment (szenius) [inline src/functionHandlers/proxy/transformers/bodyTransformer.ts:15]: Looks like we will be doing the hashing twice, once in `idMapper` and once here. Should we consolidate and only do it once?

Comment (szenius) [inline src/functionHandlers/proxy/transformers/bodyTransformer.ts:15]: Looks like we will be doing the hashing twice, once in idMapper and once here. Should we consolidate and only do it once?

Comment (ilhaoni) [reply src/functionHandlers/proxy/transformers/bodyTransformer.ts:15]: yeah, lets do it!

Comment (ilhaoni) [reply src/functionHandlers/proxy/transformers/bodyTransformer.ts:14]: updated~ :D

**Pull request merged**